test/bio_tfo_test.c

```diff
@@ -0,0 +1,418 @@
+            addr = &((struct sockaddr_in6*)&sstorage)->sin6_addr;
+            addrlen = sizeof(((struct sockaddr_in6*)&sstorage)->sin6_addr);
+            break;
+        default:
+            goto err;
+    }
+
+    if(!TEST_true(BIO_ADDR_rawmake(baddr, sstorage.ss_family, addr, addrlen, port)))
+        goto err;
+
+    /* CLIENT SOCKET */
+    if (!TEST_int_ge(cfd = BIO_socket(ai->ai_family, ai->ai_socktype, ai->ai_protocol, 0), 0))
+        goto err;
+
+    /* FIRST ACCEPT: no connection should be established */
+    sfd = BIO_accept_ex(afd, NULL, 0);
+    if (sfd == -1) {
+        sockerr = get_last_socket_error();
+        /* Note: Windows would hit WSAEWOULDBLOCK */
+        if (sockerr != EAGAIN) {
+            BIO_printf(bio_err, "Error: failed without EAGAIN\n");
+            goto err;
+        }
+    } else {
+        BIO_printf(bio_err, "Error: accepted unknown connection\n");
+        goto err;
+    }
+
+    /* CONNECT ATTEMPT: different behavior based on TFO support */
+    if (!BIO_connect(cfd, baddr, client_flags)) {
+        sockerr = get_last_socket_error();
+        if (sockerr == EOPNOTSUPP) {
+            BIO_printf(bio_err, "Skip: TFO not enabled/supported for client\n");
+            goto success;
+        } else {
+            /* Note: Windows would hit WSAEWOULDBLOCK */
+            if (sockerr != EINPROGRESS) {
+                BIO_printf(bio_err, "Error: failed without EINPROGRESS\n");
+                goto err;
+            }
+        }
+    }
+
+    /* macOS needs some time for this to happen, so put in a select */
+    if (!TEST_int_ge(BIO_socket_wait(afd, 1, time(NULL)+2), 0)) {
+        sockerr = get_last_socket_error();
+        BIO_printf(bio_err, "Error: socket wait failed\n");
+        goto err;
+    }
+
+    /* SECOND ACCEPT: if TFO is supported, this will still fail until data is sent */
+    sfd = BIO_accept_ex(afd, NULL, 0);
+    if (sfd == -1) {
+        sockerr = get_last_socket_error();
+        /* Note: Windows would hit WSAEWOULDBLOCK */
+        if (sockerr != EAGAIN) {
+            BIO_printf(bio_err, "Error: failed without EAGAIN\n");
+            goto err;
+        }
+    } else {
+        if (idx == 0)
+            BIO_printf(bio_err, "Success: non-TFO connection accepted without data\n");
+        else if (idx == 1)
+            BIO_printf(bio_err, "Ignore: connection accepted before data, possibly no TFO cookie, or TFO may not be enabled\n");
+        else if (idx == 4)
+            BIO_printf(bio_err, "Success: connection accepted before data, client TFO is disabled\n");
+        else
+            BIO_printf(bio_err, "Warning: connection accepted before data, TFO may not be enabled\n");
+        goto success;
+    }
+
+    /* SEND DATA: this should establish the actual TFO connection */
+#ifdef OSSL_TFO_SENDTO
+    if (!TEST_int_ge(sendto(cfd, SOCKET_DATA, SOCKET_DATA_LEN, OSSL_TFO_SENDTO,
+                            (struct sockaddr*)&sstorage, slen), 0)) {
```

Comment:
Nit: space before `*`

Reply:
So, AFAICT, there's no coding specification for casts. The <space> is not strictly necessary. But the reasoning is that by putting the `*` next to the type, it's a bit more obvious _in declarations_ that it's a pointer-type. Of course, this only works well when there's a single declaration per line. And that's why I tend to do that in casts (because of how I do declarations).

Reply:
We, throughout the code use space before `*` both in casts and declarations. So thank you for changing this.